### PR TITLE
[16.0][MIG] account_invoice_export

### PR DESCRIPTION
**Pull request by TDu**
Along with the migration, I split the module to make the dependency on queue job optional. See commits for details.

### CODE REVIEW

**Review by sebalix — APPROVED** [review]
Fixup to squash, LG

File: account_invoice_export/models/account_move.py [hunk 1]
```diff
@@ -65,12 +65,13 @@ def _export_invoice(self):
         file_data = self._get_file_for_transmission_method()
         headers = self.transmit_method_id.get_transmission_http_header()
         url = self.transmit_method_id.get_transmission_url()
-        res = requests.post(url, headers=headers, files=file_data)
+        res = requests.post(url, headers=headers, files=file_data, timeout=5)
```
Comment (grindtildeath) [inline account_invoice_export/models/account_move.py]: Isn't 5 seconds a bit low? Looking into odoo code it's generally between 15 and 30s. Could be good to add a parameter (a TODO in the code is also fine while working on the mig)

File: account_invoice_export/tests/common.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+from odoo.tests.common import TransactionCase
```
Comment (grindtildeath) [inline account_invoice_export/tests/common.py:3]: Please add header

**Review by grindtildeath — APPROVED** [review]
CI to fix but LG

File: account_invoice_export/tests/common.py (hunk 2 above)
Comment (TDu) [reply account_invoice_export/tests/common.py:3]: done

File: account_invoice_export/models/account_move.py (hunk 1 above)
Comment (TDu) [reply account_invoice_export/models/account_move.py]: I changed it to 10.
And also left a comment to improve with configuaration.

**Comment by TDu** [comment]
@simahawk  This one is ready to be merged IMO

**Pull request merged**